Code/Geometry/UniformGrid3D.cpp

```diff
@@ -154,6 +154,7 @@ Point3D UniformGrid3D::getGridPointLoc(unsigned int pointId) const {
   }
   Point3D res;
   res.x = (pointId % d_numX) * d_spacing;
+  // the rounding here is intentional, we want the coordinates of a grid point
```

Comment:
Can you remove the PVS warning by doing an explicit cast or an rdcast?

Reply:
```
res.x = static_cast<double>((pointId % d_numX)) * d_spacing;
res.y = statiic_cast<double>((pointId / (d_numX * d_numY))) * d_spacing;
```

Reply:
Note that this is slightly different than your fix as we cast the entire integer operations to a double which is being done implicitly in the original code.

Reply:
The warning wasn't about a missing cast. It was about the fact that the operation as written was losing precision. That's actually by design (though I didn't realize that at first).